### DISCUSSION TITLE
Support deeplinks to GitHub Mobile from the Slack mobile app

### DISCUSSION THREAD

**Discussion by tobydennison** [community/community#129114] [Mobile]
### Select Topic Area

Product Feedback

### Body

# Feature request

Support deeplinks into the GitHub mobile app from the Slack mobile app.

**My Platform:** iOS (iPhone).

**Problem:** Currently, clicking a link to a pull request (for example) in the Slack mobile app takes me to github.com in an in-app Safari browser where I have to sign in. Consequently, I have to manually navigate to the repository and then the pull request. This makes code review on the go a little more difficult.

**Wish:** I would like a GitHub link on Slack, such as to a PR I want to review, to deeplink to that content in the GitHub mobile app. 

## Response from Slack support (19/06/2024)

'The deep link functionality has to be built out by Github to recognize the deep link. Slack already has a deep link implementation set up for the Slack Client but you would have to reach out to Github to implement this on their end.'

**Comment by esahmed21** [comment]
Hi @tobydennison 👋 

Thank you for reaching out to us! Could you please share the app version you're using right now so we can help you more?

**Reply by tobydennison** [reply]
Hi, @esahmed21. Certainly.

- GitHub Mobile (iOS): 1.165.0 (133064262)
- Slack Mobile (iOS): 24.06.30 (441414)

**Reply by esahmed21** [reply]
Thanks for answering back @tobydennison.

Two things to try out:
- If you navigate to Profile, Settings, External Links. Could you make sure that In-App option is selected?
- Long tap on the GitHub link in slack, select open, and choose to open in Github app. Would that open the link in the app?  

Here is the document for universal links on iOS for reference: https://docs.github.com/en/get-started/using-github/github-mobile#managing-universal-links-for-github-mobile-on-ios

If any of these not working with you, let me know and i will create internal issue for the team to investigate further.

**Reply by tobydennison** [reply]
Hi, again. @esahmed21 

Thanks for that information. I now have links opening from Slack to GitHub, but the opt-in path isn't obvious.

Here's how I got it working.

1. Switched my preference to 'In-App' links versus 'Safari' within 'External Links' of the GitHub Mobile app settings. That did not immediately open links in the mobile app. They continued to open in Safari, but I could tap the 'OPEN' banner (screenshot below) to open the link in GitHub Mobile.

![image](https://github.com/community/community/assets/43125373/bfddf3e0-ce22-44e1-a39f-7bc84fd1fac2)

2. I copied a PR link to the iOS Notes app and was then able to long-press to see other options, such as 'Open in GitHub'. Tapping the link opened the GitHub Mobile app. You cannot see that list of long-press options in the Slack mobile app; long-pressing brings up their own reduced list (screenshot below).

![image](https://github.com/community/community/assets/43125373/98f6d894-7ec4-4c5c-b8a0-6d5a0a5c1d56)

3. To complete the test, I clicked a PR link in Slack and it took me straight to the PR in the GitHub Mobile app. 🎉

I think some additional documentation could help here, perhaps suggesting the Notes app trick I did.

I think it is a neat idea to be able to toggle "open in Safari/GitHub Mobile" depending on a normal link click or a long-press, but I think it confuses the process when you have an explicit opt-in setting within the GitHub app. In my case, I enabled 'In-App' in the GitHub Mobile app but, because of the (I suspect) additional way to toggle the link behaviour, it did not work immediately.

**Comment by H3RI3R** [comment]
Support deeplinks to GitHub Mobile from the Slack mobile app
To support deeplinks to GitHub Mobile from the Slack mobile app, you can follow these steps:

Install the GitHub app on your mobile device and log in with your GitHub credentials.
Open the Slack app and navigate to the channel where you want to receive GitHub notifications.
Tap the gear icon next to the channel name and select “Add apps.”
Search for “GitHub” in the search bar, then select “Add to Slack.”
Follow the prompts in the Slack app directory to authorize access to your GitHub account and select the repositories you want to integrate.
Once you have authorized access, you can customize the settings to choose which types of notifications you want to receive and how they should appear in Slack.
It’s important to note that deeplinks work only with default URLs in the format of WeSupport.freshdesk.com. If the link is not in this format, it may not work as expected.

In addition, you can also use the GitHub app to receive notifications for specific events, such as new issues or pull requests. To do this, follow these steps:

Open the GitHub app and navigate to the repository you want to receive notifications for.
Tap the three dots in the top right corner of the screen and select “Notifications.”
Tap “Edit” next to “Notifications” and select the types of notifications you want to receive.
Tap “Done” to save your changes.
In the Slack app, navigate to the channel where you want to receive notifications and type /github followed by the name of the repository you want to receive notifications for. For example, /github my-repo.
This will open the GitHub app and display the relevant information. You can then use the GitHub app to view the issue or pull request and take action as needed.

**Reply by Youcanttellitsme1** [reply]
I was hoping someone could help me. I believe my stepson has access to my phone
